src/extension/app-schema/app-schema-test/src/test/java/org/geoserver/test/LayersNamesAliasingTests.java

```diff
@@ -0,0 +1,233 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.custommonkey.xmlunit.XpathEngine;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.w3c.dom.Document;
+
+/**
+ * Tests that validate that layer names aliasing works correctly for App-Schema defined feature
+ * types, i.e. that is possible to create a layer with a name different from the App-Schema feature
+ * type.
+ */
+public final class LayersNamesAliasingTests extends AbstractAppSchemaTestSupport {
+
+    // xpath engines used to check WFS responses
+    private XpathEngine WFS11_XPATH_ENGINE;
+    private XpathEngine WFS20_XPATH_ENGINE;
+
+    private static Path testFolderPath;
+    private static Path mappingsFolderPath;
+
+    @BeforeClass
+    public static void prepare() throws IOException {
+        testFolderPath = Files.createTempDirectory(Paths.get("target/test-classes"), "layernames");
+        File srcDir = new File("target/test-classes/test-data/stations/layerNamesTest");
+        File destDir = Paths.get(testFolderPath.toString(), "layerNamesTest").toFile();
+        FileUtils.copyDirectory(srcDir, destDir);
+        mappingsFolderPath = destDir.toPath();
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        Catalog catalog = getCatalog();
+
+        FeatureTypeInfo info = catalog.getFeatureTypeByName("st_gml31", "lyr_Station_gml31");
+        info.setEnabled(true);
+        info.setNativeName("Station_gml31");
+        catalog.save(info);
+
+        info = catalog.getFeatureTypeByName("st_gml32", "lyr_Station_gml32");
+        info.setEnabled(true);
+        info.setNativeName("Station_gml32");
+        catalog.save(info);
+    }
+
```

Comment:
This is what I would expect to see:

- testAliasedNameWfsGetFeature11
- testAliasedNameWfsGetFeature20
- testAliasedNameWfsGetFeaturePost11
- testAliasedNameWfsGetFeaturePost20
- testLocalWorkspaceNoPrefixWfsGetFeature11
- testLocalWorkspaceNoPrefixWfsGetFeature20
- testLocalWorkspaceNoPrefixWfsGetFeaturePost11
- testLocalWorkspaceNoPrefixWfsGetFeaturePost20

I find that having the same tests that only differ on the WFS version used side by side helpful.

Reply:
Good, done.